sdk/include/opentelemetry/sdk/metrics/exporter.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include "opentelemetry/sdk/metrics/record.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace metrics
+{
+/**
+ * ExportResult is returned as result of exporting a batch of records.
+ */
+enum class ExportResult
+{
+  /**
+   * Batch was successfully exported.
+   */
+  kSuccess = 0,
+  /**
+   * Exporting failed. The caller must not retry exporting the same batch; the
+   * batch must be dropped.
+   */
+  kFailure = 1,
+
+  /**
+   * The collection does not have enough space to receive the export batch.
+   */
+  kFailureFull = 2,
+
+  /**
+   * The export() function was passed an invalid argument.
+   */
+  kFailureInvalidArgument = 3,
+};
+/**
+ * MetricsExporter defines the interface that protocol-specific span exporters must
+ * implement.
+ */
+class MetricsExporter
+{
+public:
+  virtual ~MetricsExporter() = default;
+
+  /**
+   * Exports a vector of Records. This method must not be called
+   * concurrently for the same exporter instance.
+   * @param records a vector of unique pointers to metric records
+   */
+  virtual ExportResult Export(const std::vector<Record> &records) noexcept = 0;
+
+  /**
+   * In the Metrics specification, there is no Shutdown function required for exporters
+   * The Shutdown function can be implemented within exporters that wish to have one,
+   * but will not be enforced through this header
+   */
```

Comment:
This doesn't make sense and we should remove it.

The `MetricsExporter` is passed on to the `Controller` as a `std::unique_ptr` (see [here](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/sdk/include/opentelemetry/sdk/metrics/controller.h#L30)). With that design, the user doesn't have the possibility to call any `Shutdown` method on the `MetricsExporter`, only the `Controller` could do so (but cannot if the interface doesn't have a `Shutdown` method).

I recommend removing this comment for now. We can extend the exporter interface by an optional shutdown function later on.

Reply:
This comment is originally included within the current `exporter.h` file, but I agree; will remove this comment for now.